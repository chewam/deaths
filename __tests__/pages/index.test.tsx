/**
 * @jest-environment jsdom
 */
import messages from "@/lang/fr.json"
import { IntlProvider } from "react-intl"
import { render } from "@testing-library/react"
import { ResizeObserver as ResizeObserverPolyfill } from "@juggle/resize-observer"

import Page from "../../src/pages/index"

if (typeof window !== "undefined") {
  window.ResizeObserver = window.ResizeObserver || ResizeObserverPolyfill
}

test("Page snapshot: index", () => {
  jest.mock("next/router", () => ({
    useRouter() {
      return {
        locale: "fr",
        defaultLocale: "fr",
      }
    },
  }))

  const { asFragment } = render(
    <IntlProvider locale="fr" messages={messages}>
      <Page />
    </IntlProvider>
  )
  expect(asFragment()).toMatchSnapshot()
})
