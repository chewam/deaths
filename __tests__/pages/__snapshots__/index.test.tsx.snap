// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page snapshot: index 1`] = `
<DocumentFragment>
  <div
    class="dashboard"
  >
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2022
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            67 813 396
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2021
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            67 063 703
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2020
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            67 063 703
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2019
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            66 977 703
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2018
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            66 883 761
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2017
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            66 774 482
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2016
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            66 602 645
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2015
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            66 422 469
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2014
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            66 130 873
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2013
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            65 564 756
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2012
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            65 241 241
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2011
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            64 933 400
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2010
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            64 612 939
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2009
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            64 304 500
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2008
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            63 961 859
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2007
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            63 600 690
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2006
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            63 186 117
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2005
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            62 730 537
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2004
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            62 251 062
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2003
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            61 824 030
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2002
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            61 385 070
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2001
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            60 941 410
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
    <div
      class="tile"
    >
      <div
        class="left"
      >
        <h3>
          <div
            class="year"
          >
            2000
          </div>
          <i
            class="ri-arrow-right-up-line"
            title="mortality raise"
          />
        </h3>
        <div
          class="stat"
        >
          <div
            class="value below"
          >
            0%
          </div>
          <div
            class="label"
          >
            Taux de mortalité
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            0
          </div>
          <div
            class="label"
          >
            Nombre de décès
          </div>
        </div>
        <div
          class="stat"
        >
          <div
            class="value"
          >
            60 508 150
          </div>
          <div
            class="label"
          >
            Population totale
          </div>
        </div>
      </div>
      <div
        class="right"
      />
    </div>
  </div>
</DocumentFragment>
`;
